web-admin/src/routes/[organization]/[project]/+layout.svelte

```diff
@@ -1,34 +1,20 @@
 <script lang="ts">
+  import { goto } from "$app/navigation";
   import { page } from "$app/stores";
   import RuntimeProvider from "@rilldata/web-common/runtime-client/RuntimeProvider.svelte";
-  import { createAdminServiceGetProject } from "../../../client";
+  import { useProjectRuntime } from "../../../components/projects/selectors";
 
-  $: projRuntime = createAdminServiceGetProject(
+  $: projRuntime = useProjectRuntime(
     $page.params.organization,
-    $page.params.project,
-    {
-      query: {
-        // Proactively refetch the JWT because it's only valid for 1 hour
-        refetchInterval: 1000 * 60 * 30, // 30 minutes
-        select: (data) => {
-          // There may not be a prodDeployment if the project was hibernated
-          if (!data.prodDeployment) {
-            return;
-          }
-
-          return {
-            // Hack: in development, the runtime host is actually on port 8081
-            host: data.prodDeployment.runtimeHost.replace(
-              "localhost:9091",
-              "localhost:8081"
-            ),
-            instanceId: data.prodDeployment.runtimeInstanceId,
-            jwt: data?.jwt,
-          };
-        },
-      },
-    }
+    $page.params.project
   );
+
+  $: isRuntimeHibernating = $projRuntime.isSuccess && !$projRuntime.data;
+
+  $: if (isRuntimeHibernating) {
+    // Redirect any nested routes (notably dashboards) to the project page
+    goto(`/${$page.params.organization}/${$page.params.project}`);
+  }
```

Comment:
If already on the project page, is `goto` then a no-op or could this lead to some weird re-render cycles?